lib/node_modules/@stdlib/array/base/mskfilter-map/lib/assign.js

```diff
@@ -0,0 +1,245 @@
+* @param {Object} mask - mask array object
+* @param {Collection} out - real-valued floating-point output array view
+* @param {integer} stride - output array stride
+* @param {NonNegativeInteger} offset - output array offset
+* @param {Function} clbk - function to apply
+* @param {*} [thisArg] - function execution context
+* @returns {Collection} output array view
+*
+* @example
+* var Float64Array = require( '@stdlib/array/float64' );
+* var arraylike2object = require( '@stdlib/array/base/arraylike2object' );
+*
+* var x = new Float64Array( [ 1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0 ] );
+* var mask = [ 0, 1, 0, 1 ];
+*
+* var out = new Float64Array( 8 );
+*
+* function mappingFunction( x ) {
+*     return x * 10.0;
+* }
+*
+* var arr = complex( x, arraylike2object( mask ), out, 1, 0, mappingFunction );
+* // returns <Float64Array>[ 30.0, 40.0, 70.0, 80.0, 0.0, 0.0, 0.0, 0.0 ]
+*/
+function complex( x, mask, out, stride, offset, clbk, thisArg ) {
+	var mdata;
+	var mget;
+	var io;
+	var so;
+	var i;
+	var j;
+
+	mdata = mask.data;
+	mget = mask.accessors[ 0 ];
+
+	so = stride * 2; // note: multiply by 2, as real-valued array consists of interleaved real and imaginary components
+	io = offset * 2;
+	for ( i = 0; i < mdata.length; i++ ) {
+		if ( mget( mdata, i ) ) {
+			j = i * 2;
+			out[ io ] = clbk.call( thisArg, x[ j ] );
+			out[ io+1 ] = clbk.call( thisArg, x[ j+1 ] );
+			io += so;
+		}
+	}
+	return out;
+}
+
+
+// MAIN //
+
+/**
+* Applies a mask to a provided input array & apply mapping function and assigns unmasked values to elements in a provided output array.
+*
+* @param {Collection} x - input array
+* @param {Collection} mask - mask array
+* @param {Collection} out - output array
+* @param {integer} stride - output array stride
+* @param {NonNegativeInteger} offset - output array offset
+* @param {Function} clbk - function to apply
+* @param {*} [thisArg] - function execution context
+* @returns {Collection} output array
+*
+* @example
+* var x = [ 1, 2, 3, 4 ];
+* var mask = [ 0, 1, 0, 1 ];
+*
+* function mappingFunction( x ) {
+*     return x * 10;
+* }
+*
+* var out = [ 0, 0 ];
+* var arr = assign( x, mask, out, 1, 0, mappingFunction );
+* // returns [ 20, 40 ]
+*
+* var bool = ( arr === out );
+* // returns true
+*/
+function assign( x, mask, out, stride, offset, clbk, thisArg ) {
+	var xo;
+	var mo;
+	var oo;
+
+	xo = arraylike2object( x );
+	mo = arraylike2object( mask );
+	oo = arraylike2object( out );
+	if (
+		xo.accessorProtocol ||
+		mo.accessorProtocol ||
+		oo.accessorProtocol
+	) {
+		// Note: we only explicitly support complex-to-complex, as this function should not be concerned with casting rules, etc. That is left to userland...
+		if (
+			isComplexDataType( xo.dtype ) &&
+			isComplexDataType( oo.dtype )
+		) {
+			complex( reinterpret( x, 0 ), mo, reinterpret( out, 0 ), stride, offset, clbk, thisArg ); // eslint-disable-line max-len
+			return out;
+		}
```

Comment:
We can omit this branch. Because we are supporting an arbitrary callback from userland, we cannot do tricks to speed things up. We need to materialize a complex number for each invocation.

Reply:
If we omit this branch then test case for complex type array is not valid (needed to be removed).
Should I omit this part or not?? @kgryte

Reply:
The tests for complex typed arrays need to be updated in order to accommodate the omitted branch. In `mskfilter`, we could handle complex arrays specially, but, for this package, they are just another accessor array.

Reply:
@kgryte updated tests for complex typed array.